js/ui/shellEntry.js

```diff
@@ -184,6 +184,26 @@ var OverviewEntry = GObject.registerClass({
         global.stage.connect('notify::key-focus', this._onStageKeyFocusChanged.bind(this));
     }
 
+    _updateSearchEntryText() {
+        let entryText;
+        let internetProvider = InternetSearch.getInternetSearchProvider();
+
+        if (internetProvider &&
+            this._parentalControlsManager.shouldShowApp(internetProvider.appInfo)) {
```

Comment:
To be 100% safe here we should probably check if `internetProvider.appInfo` is valid as done by `Search._registerProvider` - although I think it is always valid for internet providers.

Reply:
Is there logic in the internet search provider which checks whether it should be active or not? (If not, what prevents internet search results being shown anyway?) If so, this seems to duplicate it.

Reply:
The logic is handled in `Search._registerProvider()` which gets called when parental controls change (the search providers are unregistered and registered again). The `_registerProvider()` code checks the `parentalControlsManager` object if the provider should be active. Maybe this logic could be moved to the search providers but probably done in a separate task.

Reply:
Addressed.